lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -1335,12 +1335,20 @@ namespace Js
             EngineInterfaceObject* nativeEngineInterfaceObj = scriptContext->GetLibrary()->GetEngineInterfaceObject();
             if (nativeEngineInterfaceObj)
             {
-                IntlEngineInterfaceExtensionObject* intlExtensionObject = static_cast<IntlEngineInterfaceExtensionObject*>(nativeEngineInterfaceObj->GetEngineExtension(EngineInterfaceExtensionKind_Intl));
+                IntlEngineInterfaceExtensionObject* intlExtensionObject = static_cast<IntlEngineInterfaceExtensionObject*>(
+                    nativeEngineInterfaceObj->GetEngineExtension(EngineInterfaceExtensionKind_Intl));
                 if (args.Info.Count == 2)
                 {
                     auto undefined = scriptContext->GetLibrary()->GetUndefined();
                     CallInfo toPass(callInfo.Flags, 7);
-                    return intlExtensionObject->EntryIntl_CompareString(function, toPass, undefined, pThis, pThat, undefined, undefined, undefined, undefined);
+                    ThreadContext *threadContext = scriptContext->GetThreadContext();
+                    return threadContext->ExecuteImplicitCall(function, ImplicitCall_Accessor,
+                        [scriptContext, threadContext, intlExtensionObject, function, toPass, undefined, pThis, pThat]() -> Var
```

Comment:
Quick note that you're capturing all of these variables by copy. Consider whether capture-by-reference would be better for any of these. Also, unless you feel that it's important to limit the list of captures, you can capture everything with `[=]` (copy) or `[&]` (reference).

Reply:
Also looks like you don't need scriptContext at all.

Reply:
I didn't want to capture the pointers by reference, and i think most/all of these are pointers. With that being said, I didnt want to capture everything by copy since I didn't necessarily know which copy constructors would run as a result. You're right that I don't need scriptContext anymore, i did before.